core/src/sigverify_stage.rs

```diff
@@ -206,13 +236,23 @@ impl SigVerifyStage {
             timing::timestamp(),
             num_packets,
         );
+
         let mut discard_time = Measure::start("sigverify_discard_time");
         if num_packets > MAX_SIGVERIFY_BATCH {
             Self::discard_excess_packets(&mut batches, MAX_SIGVERIFY_BATCH);
         }
+        let excess_fail = count_discard_in_batches(&batches);
```

Comment:
might be faster to do all this counting within the operations themselves when you are already loading/walking the packet buffers. to access `.discard()` will be strided access.